### PR TITLE
DBZ-2948 Oracle BLOB/CLOB, LONG RAW, and RAW data type support

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-2948

This PR is based on the PR for DBZ-3322.  This provides support for BLOB/CLOB data types.  

Both data types can be decoded from the LogMiner view in various ways depending entirely on the size of the LOB value.  If the length of the LOB is small enough that it can be stored in the data page of the row, then LogMiner will only see DML operations for the LOB values, but these LOB operations need to be combined into a single emitted event.

For example, if a CLOB field is to store something small such as `Acme Co`, then the LogMiner event stream would be
```
SCN TYPE   SQL
--- ------ ------------
100 INSERT insert into table ("ID","DATA") values ('1', EMPTY_CLOB());
101 UPDATE update table set "DATA" = 'Acme Co.' WHERE "ID" = '1';
```
In the above example, the code changes in this PR would in turn emit this event as a single INSERT where the after state has the values for both "ID" and "DATA" columns.

In the event that the LOB field exceeds what can be stored in the row's data page, then LogMiner will materialize the LOB as a series of `SEL_LOB_LOCATOR`, `LOB_WRITE`, and optionally an `LOB_TRIM` operation sequence per LOB-column.  Also, if the table contains any non-LOB fields beyond the primary key, an additional synthetic `UPDATE` operation will also exist in the the view.  All of these events need to be combined so that if the event series is for a new record, we emit a single Create event.  If the event series is for an update to an existing record, we emit a single Update event.

This still lacks the following changes, which are yet to come:

- [x] Documentation updates for all new supported data types
- [x] ~~Support for `LONG`, `LONG RAW` and `RAW` data types for both adapters~~ (Not applicable)
- [x] XStream support for `CLOB` and `BLOB` data types

### CODE REVIEW

**Comment by Naros** [comment]
@ant0nk has identified a bug in testing this pull request.  Once I have a fix I'll push the change and have him re-test.

**Comment by Naros** [comment]
@ant0nk I've added a fix for the `SelectLobParser` exception you provided as well as a corner case with incorrectly updating the offset's SCN using the new overlapped mining approach.  Any feedback on any additional errors would be great otherwise we'll plan to include this in this week's next 1.6 release.

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Applied, thx a lot, @Naros!

**Comment by gunnarmorling** [comment]
Hey @ant0nk, so this will be part of today's (hopefully) 1.6.0.Beta1 release; would love to get your feedback on this after further testing. Thank you so much!

**Comment by ant0nk** [comment]
Hi, @gunnarmorling, ok, I'll check it.

**Comment by gunnarmorling** [comment]
Thanks, @ant0nk!

**Comment by ant0nk** [comment]
@gunnarmorling, I tested the 1.6.0.Beta1 release in my environment and didn't encounter any errors .

**Comment by gunnarmorling** [comment]
Hey Anton, excellent, that's very encouraging to hear! Thank you so much
for testing.

>

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java [hunk 1]
```diff
@@ -357,25 +356,33 @@ public Scn getCurrentScn() throws SQLException {
...
+    public byte[] getHexToRawByteArray(String hexToRawValue) throws SQLException {
+        final String data;
+        if (hexToRawValue.startsWith("HEXTORAW('") && hexToRawValue.endsWith("')")) {
+            data = hexToRawValue.substring(10, hexToRawValue.length() - 2);
+        }
+        else {
+            data = hexToRawValue;
+        }
+
+        return prepareQueryAndMap("select HEXTORAW(?) FROM DUAL", ps -> ps.setString(1, data), rs -> {
```
Comment (TAregger) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java:378]: Hey @Naros, sorry to hijack this already merged PR but I just stumbled over this code and noticed that it was recently committed, so I thought I would use this PR to comment.

This basically sends every LOB (or LOB chunk) again to the db server and reads back the converted value. So in the end each LOB gets transferred between database and connector 3 times. This is potentially a huge amount of data sent over the wire as we are dealing with LOBs here. Is there a reason why `oracle.sql.RAW#hexString2Bytes` was not used here?

Comment (gunnarmorling) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java:378]: Captured this in https://issues.redhat.com/browse/DBZ-3556, so to keep track of this. Thanks, @TAregger!
